pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java

```diff
@@ -409,21 +411,19 @@ public Object visit(ASTName node, Object data) {
 
         JavaTypeDefinition previousType;
 
-        // TODO: static method invocation
-        if (dotSplitImage.length == 1 && astArguments != null) { // method
+        if (node.getType() != null) { // static field or method
```

Comment:
I'm lost here... who is setting this on a static access? I don't see it being set in this method up to this point, and `ASTName` has no children....